src/util/metrics.ts

```diff
@@ -26,32 +118,245 @@ export default class Metrics_App {
+			// if (!req.route && res && res.statusCode === 404) {
+			route = 'N/A';
+		}
+
+		return route;
+	}
+
+	private blocksControllerMetrics(req: Query, res: Response, end: () => number) {
+		const body = res.locals.body as Body | Body[];
+
+		if (req.params?.number && !Array.isArray(body)) {
+			const extrinscs = body.extrinsics ? body.extrinsics.length : 0;
+
+			const extrinsics_per_second = this.metrics['sas_extrinsics_per_second'] as client.Histogram;
+			const seconds = end();
+
+			extrinsics_per_second
+				.labels({ method: req.method, route: this.getRoute(req), status_code: res.statusCode })
+				.observe(extrinscs / seconds);
+
+			const extrinsics_per_block = this.metrics['sas_extrinsics_per_block'] as client.Histogram;
+
+			const blocks = 1;
+			extrinsics_per_block
+				.labels({ method: req.method, route: this.getRoute(req), status_code: res.statusCode })
+				.observe(extrinscs / blocks);
+
+			const seconds_per_block = this.metrics['sas_seconds_per_block'] as client.Histogram;
+			seconds_per_block
+				.labels({ method: req.method, route: this.getRoute(req), status_code: res.statusCode })
+				.observe(seconds / blocks);
+		}
+
+		if (req.query?.range) {
+			let totExtrinsics = 0;
+			if (Array.isArray(body)) {
+				totExtrinsics = body.reduce((current: number, block: { [x: string]: unknown }) => {
+					const extrinsics = block['extrinsics'];
+
+					if (Array.isArray(extrinsics)) {
+						return current + extrinsics.length;
+					}
+
+					return current;
+				}, 0);
+			} else {
+				const extrinsics = body['extrinsics'];
+				if (Array.isArray(extrinsics)) {
+					totExtrinsics = extrinsics.length;
+				}
+			}
+
+			const extrinsics_in_request = this.metrics['sas_extrinsics_in_request'] as client.Histogram;
+			extrinsics_in_request
+				.labels({ method: req.method, route: this.getRoute(req), status_code: res.statusCode })
+				.observe(totExtrinsics);
+
+			const extrinsics_per_second = this.metrics['sas_extrinsics_per_second'] as client.Histogram;
+			const seconds = end();
+			extrinsics_per_second
+				.labels({ method: req.method, route: this.getRoute(req), status_code: res.statusCode })
+				.observe(totExtrinsics / seconds);
+
+			const extrinsics_per_block = this.metrics['sas_extrinsics_per_block'] as client.Histogram;
+			const blocks = Array.isArray(body) ? body.length : 1;
+			extrinsics_per_block
+				.labels({ method: req.method, route: this.getRoute(req), status_code: res.statusCode })
+				.observe(totExtrinsics / blocks);
+
+			const seconds_per_block = this.metrics['sas_seconds_per_block'] as client.Histogram;
+			seconds_per_block
+				.labels({ method: req.method, route: this.getRoute(req), status_code: res.statusCode })
+				.observe(seconds / blocks);
+		}
+	}
+
+	preMiddleware() {
+		return (req: Query, res: Response, next: () => void) => {
+			const tot_requests = this.metrics['sas_total_requests'] as client.Counter;
+
+			// request count metrics
+			tot_requests.inc();
```

Comment:
```suggestion
			if (req.originalUrl != '/favicon.ico') {
				tot_requests.inc();
			}
```
This is related to the comments below. Without this check and with every requested endpoint we will see the `sas_total_requests`  counter increase by 2 (it counts the request from the user plus the favicon request) instead of 1.